components/org.wso2.carbon.micro.integrator.extensions/org.wso2.carbon.micro.integrator.management.apis/src/main/java/org/wso2/carbon/micro/integrator/management/apis/TemplateResource.java

```diff
@@ -0,0 +1,204 @@
+        Map<String, TemplateMediator> sequenceTemplateMap = synapseConfiguration.getSequenceTemplates();
+        JSONObject jsonBody = new JSONObject();
+        JSONArray endpointTemplateList = new JSONArray();
+        JSONArray sequenceTemplateList = new JSONArray();
+        jsonBody.put(ENDPOINT_TEMPLATE_LIST, endpointTemplateList);
+        jsonBody.put(SEQUENCE_TEMPLATE_LIST, sequenceTemplateList);
+
+        if (Objects.nonNull(endpointTemplateMap) && !endpointTemplateMap.isEmpty()) {
+            for (Map.Entry<String, Template> templateEntry : endpointTemplateMap.entrySet()) {
+                Template endpointTemplate = templateEntry.getValue();
+                addToJsonList(jsonBody.getJSONArray(ENDPOINT_TEMPLATE_LIST), endpointTemplate.getName());
+            }
+        }
+        if (Objects.nonNull(sequenceTemplateMap) && !sequenceTemplateMap.isEmpty()) {
+            for (Map.Entry<String, TemplateMediator> templateEntry : sequenceTemplateMap.entrySet()) {
+                TemplateMediator sequenceTemplate = templateEntry.getValue();
+                addToJsonList(jsonBody.getJSONArray(SEQUENCE_TEMPLATE_LIST), sequenceTemplate.getName());
+            }
+        }
+        Utils.setJsonPayLoad(axis2MessageContext, jsonBody);
+    }
+
+    /**
+     * Sets the list of templates of a given type to the Axis2MessageContext as a JSON object
+     *
+     * @param templateType   type of the template Ex: sequence, endpoint
+     */
+    private void populateTemplateListByType(String templateType) {
+        if (ENDPOINT_TEMPLATE_TYPE.equalsIgnoreCase(templateType)) {
+            Map<String, Template> endpointTemplateMap = synapseConfiguration.getEndpointTemplates();
+            JSONObject jsonBody = Utils.createJSONList(endpointTemplateMap.size());
+            for (Map.Entry<String, Template> templateEntry : endpointTemplateMap.entrySet()) {
+                Template template = templateEntry.getValue();
+                addToJsonList(jsonBody.getJSONArray(Constants.LIST), template.getName());
+            }
+            Utils.setJsonPayLoad(axis2MessageContext, jsonBody);
+        } else if (SEQUENCE_TEMPLATE_TYPE.equalsIgnoreCase(templateType)) {
+            Map<String, TemplateMediator> sequenceTemplateMap = synapseConfiguration.getSequenceTemplates();
+            JSONObject jsonBody = Utils.createJSONList(sequenceTemplateMap.size());
+            for (Map.Entry<String, TemplateMediator> templateEntry : sequenceTemplateMap.entrySet()) {
+                TemplateMediator templateMediator = templateEntry.getValue();
+                addToJsonList(jsonBody.getJSONArray(Constants.LIST), templateMediator.getName());
+            }
+            Utils.setJsonPayLoad(axis2MessageContext, jsonBody);
+        } else {
+            log.warn("Provided template type does not exist");
+            Utils.setJsonPayLoad(axis2MessageContext, Utils.createJSONErrorObject("Invalid template type"));
```

Comment:
Shall we add the used template type to the response.

Reply:
done